mapbox/libandroid-telemetry/src/main/java/com/mapbox/services/android/telemetry/permissions/PermissionsListener.java

```diff
@@ -0,0 +1,15 @@
+package com.mapbox.services.android.telemetry.permissions;
+
+import java.util.List;
+
+/**
+ * Created by antonio on 1/11/17.
+ */
+
+public interface PermissionsListener {
+
+  void onExplanationNeeded(List<String> permissionsToExplain);
```

Comment:
Let the developer know that they need to provide an explanation to the user as indicated in [the docs](https://developer.android.com/training/permissions/requesting.html#perm-request). Basically, it covers the `ActivityCompat.shouldShowRequestPermissionRationale() == true` flow.